### PR TITLE
Mod 2 d 3 d mappers

### PR DESCRIPTION
**Pull request by mvervaec**
-Description:
The axisymmetric mappers 'axisymmetric_2d_to_3d' and 'axisymmetric_3d_to_2d' are adapted, so the user can select wither the case has to be mapped with 'wedge axisymmetric' (e.g. OpenFOAM axisymmetric) or a full 3D case (e.g. tube OpenFOAM3d-Abaqus2d). The name of the mappers in this branch to review are: 'axisymmetric_2d_to_3d_mod' and 'axisymmetric_3d_to_2d_mod'.  The flag installed to switch from 3d to wedge and vice versa is called: 'wedge'

-Tests:
 Unit tests are executed. To swap from 3d to wedge and vice versa has been tested and checked with the gui. The name of the unit tests are: 'test_axisymmetric_2d_to_3d_mod' and 'test_axisymmetric_3d_to_2d_mod' 

-Checklist:
Check if the mappers work with flag 'wedge'

### CODE REVIEW

**Review by joris13 — CHANGES_REQUESTED** [review]
- The idea is not to make a ...Mod version. The code has to be integrated in the axisymmetric_3d_to_2d class, without changing the filename, such that there is only 1 file and not a "normal version" and a "mod version"
- Do not use a bool "wedge" as parameter and then if (wedge) ... else ... in the code, but really make it one code with an angle as parameter (default 2pi) and then using -angle/2 to +angle/2 for all the code.

File: tests/mappers/test_axisymmetric_2d_to_3d.py [hunk 1]
```diff
@@ -35,16 +36,24 @@ def test_instantiation(self):
     def test_initialize(self):
         mp_name_in = 'wall_in'
         mp_name_out = 'wall_out'
+        key = 'angle'
+        if key in self.parameters['settings']:
+            self.angle = self.parameters['settings']['angle']
+            print(key + ' is set on ' + str(self.angle) + ' degrees')
+        else:
+            self.angle = 360
+            print( key + ' is set on ' + str(self.angle) + ' degrees')
```
Comment (nicolasdlss) [inline tests/mappers/test_axisymmetric_2d_to_3d.py:45]: Modify this to  `angle = self.parameters['settings'].get('angle', 360)` (I would not make angle a class variable, as you only use it locally)

File: coupling_components/mappers/axisymmetric_2d_to_3d.py [hunk 2]
```diff
@@ -22,11 +22,13 @@ def __init__(self, parameters):
         self.dir_a = dirs.index(self.settings['direction_axial'])
         self.dir_r = dirs.index(self.settings['direction_radial'])
         self.dir_3d = ({0, 1, 2} - {self.dir_a, self.dir_r}).pop()
+        self.angle  = self.settings.get('angle',360) #angle is set in degrees
```
Comment (nicolasdlss) [inline coupling_components/mappers/axisymmetric_2d_to_3d.py:25]: - The formatting in the files is a lot better, but there are still improvements possible. Using the `CTRL` + `ALT` + `L` automatically correctly formats your file. (applies to all files)

File: examples/post_processing/animate_example.py [hunk 3]
```diff
@@ -8,7 +8,7 @@
 
 # different cases to be plotted
 common_path = '../../examples/'
-case_paths = ['tube_tube_flow_tube_structure/case_results.pickle']
+case_paths = ['tube_fluent3d_abaqus2d/case_results.pickle']
```
Comment (nicolasdlss) [inline examples/post_processing/animate_example.py:11]: - _`examples/post_processing/animate_example.py`_ should not be changed

File: examples/tube_fluent3d_abaqus2d/parameters.json [hunk 4]
```diff
@@ -139,8 +139,9 @@
                     "type": "mappers.axisymmetric_3d_to_2d",
                     "settings": {
                       "direction_axial": "y",
-                      "direction_radial": "x",
-                      "n_tangential": 8
+                      "direction_radial":"x",
+	              "angle":360,
+                      "n_tangential": 10
```
Comment (nicolasdlss) [inline examples/tube_fluent3d_abaqus2d/parameters.json:144]: - _`examples/tube_fluent3d_abaqus2d/parameters.json`_ should not be changed (360 is default)

File: tests/mappers/test_axisymmetric_3d_to_2d.py [hunk 5]
```diff
@@ -15,12 +15,20 @@ def setUp(self):
...
     def test_initialize(self):
         mp_name_in = 'wall_in'
         mp_name_out = 'wall_out'
+        key = 'angle'
+        if key in self.parameters['settings']:
+            self.angle = self.parameters['settings']['angle']
+            print(key + ' is set on ' + str(self.angle) + ' degrees')
+        else:
+            self.angle = 360
+            print(key + ' is set on ' + str(self.angle) + ' degrees')
```
Comment (nicolasdlss) [inline tests/mappers/test_axisymmetric_3d_to_2d.py:31]: Modify this to angle = self.parameters['settings'].get('angle', 360) (I would not make angle a class variable, as you only use it locally)

**Review by nicolasdlss — COMMENTED** [review]
- _`examples/post_processing/animate_example.py`_ should not be changed
- _`examples/tube_fluent3d_abaqus2d/parameters.json`_ should not be changed (360 is default)
- The formatting in the files is a lot better, but there are still improvements possible. Using the `CTRL` + `ALT` + `L` automatically correctly formats your file.
- Don't forget to modify the documentation accordingly
- In fact these are not very good tests, because they use the same code as the code they test... But I understand it's hard to make good tests for this. This is also the way Toon did it. One option would be to check the number of points for each axial coordinate an see if they are equal to `n_tangential`, check that they are on a circle with the correct radius, check that the angle between them is correct and check that the angle between the furthest and the largest is correct.
- In order to also test the full circle, I would add 
```
    def test_initialize_360(self):
        self.parameters['settings'].pop('angle')
        self.test_initialize()
```
After the initialize test.

**Comment by nicolasdlss** [comment]
The relevant files from this pull request have been copied to a new pull request update mappers and test #131.

**Pull request closed without merging**